app/src/main/java/com/fastlink/zekrapp/MainActivity.kt

```diff
@@ -0,0 +1,50 @@
+package com.fastlink.zekrapp
+
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.enableEdgeToEdge
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.compositionLocalOf
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.rememberNavController
+import com.fastlink.zekrapp.appData.ZekirCategorySingleton
+import com.fastlink.zekrapp.ui.theme.ZekrAppTheme
+import com.fastlink.zekrapp.presentation.navigation.Navigation
+import com.fastlink.zekrapp.viewModel.ZekirCategoryViewModel
+
+
+val LocalNavController = compositionLocalOf<NavHostController> { error("error in navController") }
+val LocalViewModel = compositionLocalOf<ZekirCategoryViewModel> { error("error in view model") }
+
+class MainActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        enableEdgeToEdge()
+        setContent {
+            val context = LocalContext.current
+            ZekirCategorySingleton.getZekirCategoriesFromCSVFile(context)
```

Comment:
Call this in the view model upon initialization

Reply:
but i have one question you can't call this function in viewModel because need @Composable annotation for getting LocalContext

Reply:
you ca remove this parametter and set the getZekirCategoryFromCSVFile fun to @Composable function but then you ca access it in init of viewmodel

Reply:
you must do that 
1. first you add annotation `Composable` to fun and removing parameter context and set variable in function that can access it 
2. create a fun in viewmodel and call `getZekirCategoryFromCSVFIle` on it 
3.call viewmodel function in initialization of the app 


that's true , do this ?

Reply:
LocalContext is not the only way to get the context.

Reply:
i solved this problem by another way